src/server/registry/route.go

```diff
@@ -33,7 +33,8 @@ func RegisterRoutes() {
 	root := router.NewRoute().
 		Path("/v2").
 		Middleware(artifactinfo.Middleware()).
-		Middleware(v2auth.Middleware())
+		Middleware(v2auth.Middleware()).
+		Middleware(readonly.Middleware())
```

Comment:
The `/v2/xxx` requests should already be covered by the global middleware?

Reply:
yes, will remove it.